internal/rest-client/flasharray_client.go

```diff
@@ -35,16 +35,18 @@ type FAClient struct {
 	RestClient *resty.Client
 	ApiVersion string
 	XAuthToken string
+	XRequestID string
 	Error      error
 }
 
-func NewRestClient(endpoint string, apitoken string, apiversion string, uagent string, debug bool) *FAClient {
+func NewRestClient(endpoint string, apitoken string, apiversion string, uagent string, rid string, debug bool) *FAClient {
 	type ApiVersions struct {
 		Versions []string `json:"version"`
 	}
 	fa := &FAClient{
 		EndPoint:   endpoint,
 		ApiToken:   apitoken,
+		XRequestID: "",
```

Comment:
can we store the XRequestID to fa directly and reuse it for all the requests below?

Reply:
Yes, good call, updated.